### DISCUSSION TITLE
Docker-compose emqx-edge-->emqx bridge

### DISCUSSION THREAD

**Discussion by PanagB** [emqx/emqx#5405] [Q&A] [answered]
Hi,im trying to make a bridge between emqx-edge and emqx on docker-compose. I use mqttFx as clients to test the forwaded messages, but it doesnt work...bellow is the docker compose i have so far!Any idea why this does not work?

[version: '3.3'

networks:
  poc-bridge:
    external: true

services:
  mqtt-edge:
    image: emqx/emqx-edge:latest
    container_name: edge-broker
    restart: always
    environment:
      - EMQX_LOADED_PLUGINS= "emqx_bridge_mqtt"
      - EMQX_ADMIN_PASSWORD=brokerpw1
      - EMQX_BRIDGE__MQTT__AWS__ADDRESS=172.26.0.2:1883 # docker-ip:port of mqtt-cloud
      - EMQX_BRIDGE__MQTT__AWS__PROTO_VER=mqttv4     
      - EMQX_BRIDGE__MQTT__AWS__BRIDGE_MODE=true          
      - EMQX_BRIDGE__MQTT__AWS__CLEAN_START=true           
      - EMQX_BRIDGE__MQTT__AWS__USERNAME=user          
      - EMQX_BRIDGE__MQTT__AWS__PASSWORD=passw                                                       
      - EMQX_BRIDGE__MQTT__AWS__CLIENTID=bridge_aws
      - EMQX_BRIDGE__MQTT__AWS__KEEPALIVE=10s                 
      - EMQX_BRIDGE__MQTT__AWS__FORWARDS=topic/#
    networks:
      - poc-bridge   
    ports:
      - 18083:18083
      - 1883:1883
      - 8883:8883
      - 8083:8083
    depends_on:
      - mqtt-cloud
      
  mqtt-cloud:
    image: emqx/emqx:latest
    container_name: cloud-broker
    restart: always
    environment:
      - EMQX_LOADED_PLUGINS= "emqx_bridge_mqtt"
      - EMQX_ADMIN_PASSWORD=public
    networks:
      - poc-bridge
    ports:
      - 51883:1883
      - 58883:8883
      - 58083:8083
]

**Comment by zmstone** [comment]
Hi @PanagB
is it possible for you to try 4.3 instead?

**Reply by PanagB** [reply]
Hi Zmstone, thanks for your time. I've tried a few things with the version and it does not seem to be the problem!

**Comment by PanagB** [comment, marked as answer]
Problem solved. I was missing some more configurations... Bellow is the docker-compose yaml for bridging emqx-edge to emqx.

version: '3.3'

networks:
  poc-bridge:
    external: true


services:
  mqtt-edge:
    image: emqx/emqx-edge:latest
    container_name: edge-broker
    restart: always
    environment:
      - EMQX_LOADED_PLUGINS="emqx_bridge_mqtt,emqx_recon,emqx_retainer,emqx_management,emqx_dashboard"
      - EMQX_ADMIN_PASSWORD=brokerpw1
      - EMQX_BRIDGE__MQTT__AWS__START_TYPE=auto
      - EMQX_BRIDGE__MQTT__AWS__ADDRESS=172.26.0.2:1883 # docker-ip:port of mqtt-cloud                    
      - EMQX_BRIDGE__MQTT__AWS__PROTO_VER=mqttv3     
      - EMQX_BRIDGE__MQTT__AWS__BRIDGE_MODE=true        
      - EMQX_BRIDGE__MQTT__AWS__CLEAN_START=true           
      - EMQX_BRIDGE__MQTT__AWS__USERNAME=user         
      - EMQX_BRIDGE__MQTT__AWS__PASSWORD=passw                                                       
      - EMQX_BRIDGE__MQTT__AWS__CLIENTID=bridge_aws
      - EMQX_BRIDGE__MQTT__AWS__KEEPALIVE=60s
      - EMQX_BRIDGE__MQTT__AWS__FORWARD_MOUNTPOINT=bridge/aws/test/                                             
      - EMQX_BRIDGE__MQTT__AWS__FORWARDS=atlas/#
      #- EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__1__TOPIC=atlas/#                          
      #- EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__1__QOS=1                                                                                            
    networks:
      - poc-bridge   
    ports:
      - 18083:18083
      - 1883:1883
      - 8883:8883
      - 8083:8083
    depends_on:
      - mqtt-cloud
      
  mqtt-cloud:
    image: emqx/emqx:latest
    container_name: cloud-broker
    restart: always
    environment:
      - EMQX_LOADED_PLUGINS= "emqx_bridge_mqtt"
    networks:
      - poc-bridge
    ports:
      - 51883:1883
      - 58883:8883
      - 58083:8083
